UPGRADE_NOTES.md

```diff
@@ -1,4 +1,79 @@
-### Upgrading from 0.14 to 0.15
+## Upgrading from 0.16 to 0.17
+
+- Ensure you've upgraded to crystal 0.30.0
+- Upgrade Lucky CLI (homebrew)
+
+```
+brew update
+brew upgrade crystal-lang # Make sure you're up-to-date. Requires 0.30.0
+brew upgrade lucky
+```
+
+- Upgrade Lucky CLI (Linux)
+
+- Update `.crystal-version` to `0.30.0`
+
+> Remove the existing Lucky binary and follow the Linux
+> instructions in this section
+> https://luckyframework.org/guides/getting-started/installing#on-linux
+
+- Update versions in `shard.yml`
+  - Lucky should be `~> 0.17`
+- Run `shards update`
+
+### General updates
+- Rename: Action rendering method `text` to `plain_text`.
+- Update: use of `number_to_currency` now returns a String instead of writing to the view directly.
+- Delete: `config/static_file_handler.cr`. The `Lucky::StaticFileHandler` no longer has config settings.
+- Add: a new `Lucky::LogHandler` configure to `config/logger.cr`.
+```
+Lucky::LogHandler.configure do |settings|
+  # Skip logging static assets in development
+  if Lucky::Env.development?
+    settings.skip_if = ->(context : HTTP::Server::Context) {
+      context.request.method.downcase == "get" &&
+      context.request.resource.starts_with?(/\/css\/|\/js\/|\/assets\//)
+    }
+  end
+end
+```
```

Comment:
This file is extremely different than what I have in my config/logger.cr

Is this what we're recommending?

<details>

```crystal
require "file_utils"

logger =
  if Lucky::Env.test?
    # Logs to `tmp/test.log` so you can see what's happening without having
    # a bunch of log output in your specs results.
    FileUtils.mkdir_p("tmp")
    Dexter::Logger.new(
      io: File.new("tmp/test.log", mode: "w"),
      level: Logger::Severity::DEBUG,
      log_formatter: Lucky::PrettyLogFormatter
    )
  elsif Lucky::Env.production?
    # This sets the log formatter to JSON so you can parse the logs with
    # services like Logentries or Logstash.
    #
    # If you want logs like in develpoment use `Lucky::PrettyLogFormatter`.
    Dexter::Logger.new(
      io: STDOUT,
      level: Logger::Severity::INFO,
      log_formatter: Dexter::Formatters::JsonLogFormatter
    )
  else
    # For development, log everything to STDOUT with the pretty formatter.
    Dexter::Logger.new(
      io: STDOUT,
      level: Logger::Severity::DEBUG,
      log_formatter: Lucky::PrettyLogFormatter
    )
  end

Lucky.configure do |settings|
  settings.logger = logger
end

Avram.configure do |settings|
  settings.logger = logger
end
```
</details>

Reply:
I didn't want to muddle the upgrade notes. How do you do that collapse thingie? Maybe the UPGRADE_NOTES will parse that and we can show the whole file but hidden?

Reply:
Good point! I think you can use `details` https://gist.github.com/ericclemmons/b146fe5da72ca1f706b2ef72a20ac39d

Reply:
noice!